packages/mui-base/src/BadgeUnstyled/BadgeUnstyled.spec.tsx

```diff
@@ -23,7 +23,7 @@ const Badge = React.forwardRef(function Badge(
 
 const styledBadge = <BadgeUnstyled components={{ Root, Badge }} />;
 
-const PolymorphicComponentTest = () => {
+const polymorphicComponentTest = () => {
   const CustomComponent: React.FC<{ stringProp: string; numberProp: number }> = () => <div />;
```

Comment:
```suggestion
const CustomComponent: React.FC<{ stringProp: string; numberProp: number }> = () => <div />;

const PolymorphicComponentTest = () => {
```

Would this be better? I know that it produces no difference here since we are not using `polymorphicComponentTest` but it seems like a component here (or `polymorphicComponentTest` is not needed at all).

Reply:
I usually keep things specific to a test in its scope before hoisting up. That way other tests don't start to rely on it. Up to you, what you prefer.

Reply:
> I usually keep things specific to a test in its scope before hoisting up

That sounds good to me as well.

Reply:
I guess we could also disable `react/no-unstable-nested-components` for the TypeScript tests.

https://github.com/mui/material-ui/blob/101fa0d739506176bc6a3c5d82ccc03d50552f9b/.eslintrc.js#L319-L320

Reply:
It's still "invalid React". Better to write it correct everywhere.